packages/tables/src/Concerns/CanSelectRecords.php

```diff
@@ -45,7 +45,16 @@ public function getAllTableRecordsCount(): int
             return $this->records->total();
         }
 
-        return $this->getFilteredTableQuery()->count();
+        $query = $this->getFilteredTableQuery();
+
+        if (is_callable($this->getTableRecordBulkActionsEnabled())) {
+            return $query
+                ->get()
+                ->reject(fn ($item) => ! $this->getTableRecordBulkActionsEnabled()($item))
```

Comment:
Double negation is hard to read.

```suggestion
                ->filter(fn ($item): bool => $this->getTableRecordBulkActionsEnabled()($item))
```